pkg/network/protocols/http2/protocol.go

```diff
@@ -254,7 +288,47 @@ func (p *protocol) setupMapCleaner(mgr *manager.Manager) {
 		return started > 0 && (now-started) > ttl
 	})
 
-	p.mapCleaner = mapCleaner
+	p.http2InFlightMapCleaner = mapCleaner
+}
+
+func (p *protocol) setupDynamicTableMapCleaner(mgr *manager.Manager) {
+	dynamicTableMap, _, err := mgr.GetMap(dynamicTable)
+	if err != nil {
+		log.Errorf("error getting %q map: %s", dynamicTable, err)
+		return
+	}
+
+	mapCleaner, err := ddebpf.NewMapCleaner(dynamicTableMap, new(http2DynamicTableIndex), new(http2DynamicTableEntry))
+	if err != nil {
+		log.Errorf("error creating map cleaner: %s", err)
+		return
+	}
+
+	terminatedConnectionsMap := make(map[netebpf.ConnTuple]struct{})
+	mapCleaner.Clean(p.cfg.HTTP2DynamicTableMapCleanerInterval,
+		func() bool {
+			maps.Clear(terminatedConnectionsMap)
```

Comment:
would you prefer `terminatedConnectionsMap = make(...)` instead?

Reply:
Up to you – rotating the map will ensure that we'll not run into a problem of having an oversized map caused by a burst of deletions, but at the expense of more allocations, so I guess it's a pick your poison kind of situation

I'm approving the PR as it is because I think you follow-up on this in a separate PR _if needed_